bindings/wasm/src/account/storage/traits.rs

```diff
@@ -267,19 +269,20 @@ interface Storage {
   setPublishedGeneration: (did: DID, generation: Generation) => Promise<void>;
 }"#;
 
+// from https://github.com/rustwasm/wasm-bindgen/issues/2231#issuecomment-656293288
 #[allow(dead_code)]
 pub fn downcast_js_value<T: FromWasmAbi<Abi = u32>>(js_value: JsValue, classname: &str) -> Result<T, AccountError> {
   let constructor_name: js_sys::JsString = Object::get_prototype_of(&js_value).constructor().name();
   if constructor_name == classname {
     let ptr: JsValue = Reflect::get(&js_value, &JsValue::from_str("ptr"))
-      .map_err(|e| AccountError::PromiseError(format!("Invalid value: {:?} - Get ptr error: {:?}", js_value, e)))?;
+      .map_err(|e| AccountError::JsError(format!("Invalid value: {:?} - Get ptr error: {:?}", js_value, e)))?;
     let ptr_u32: u32 = ptr
       .as_f64()
-      .ok_or_else(|| AccountError::PromiseError(format!("Invalid value: {:?} - Casting error", js_value)))?
+      .ok_or_else(|| AccountError::JsError(format!("Invalid value: {:?} - Casting error", js_value)))?
```

Comment:
```suggestion
      .ok_or_else(|| AccountError::JsError(format!("unable to read `ptr` property of `JsValue` as f64")))?
```

Sorry, I missed these errors last time. I think they could be clearer.